client/__tests__/e2e/e2e.test.ts

```diff
@@ -101,6 +104,143 @@ const testURLs = {
+  });
+  test("expansion of 'Marker Gene Sets' reveals specific genesets", async ({
+    page,
+  }) => {
+    await goToPage(page, pageURLCellGuide);
+
+    // Locate and expand the 'Marker Gene Sets' header
+    const markerGeneSetsHeader = await page.locator(
+      "h5:has-text('Marker Gene Sets')"
+    );
+    await markerGeneSetsHeader.click(); // Assuming clicking will expand the section
+
+    tryUntil(
+      async () => {
+        // Assert the presence of specific genesets
+        const geneset1 = await page.locator(
+          `div[data-testid="geneset"]:has-text("enteric smooth muscle cell")`
+        );
+        const geneset2 = await page.locator(
+          `div[data-testid="geneset"]:has-text("smooth muscle fiber of ileum")`
+        );
+
+        await expect(geneset1).toBeVisible();
+        await expect(geneset2).toBeVisible();
+      },
+      { page }
+    );
+  });
+  test("expansion of 'enteric smooth muscle cell' reveals genes", async ({
+    page,
+  }) => {
+    await goToPage(page, pageURLCellGuide);
+
+    // Locate and expand the 'Marker Gene Sets' header if not already expanded
+    const markerGeneSetsHeader = await page.locator(
+      "h5:has-text('Marker Gene Sets')"
+    );
+    const chevronDownIcon = markerGeneSetsHeader.locator(
+      "svg[data-icon='chevron-down']"
+    );
+    if ((await chevronDownIcon.count()) === 0) {
+      await markerGeneSetsHeader.click();
+    }
+
+    // Locate and expand 'enteric smooth muscle cell' geneset
+    await expandGeneset("enteric smooth muscle cell - marker genes", page);
+
+    // Assert the presence of the genes div and that it contains 100 child divs
+    const genesDiv = await page.locator(`div[data-testid="gene-set-genes"]`);
+    await expect(genesDiv).toBeVisible();
+    const childDivs = await genesDiv.locator(
+      'div[data-testid*=":gene-expand"]'
+    );
+    await expect(childDivs).toHaveCount(55);
+  });
+  test("deleting 'enteric smooth muscle cell' geneset and refreshing adds it back", async ({
+    page,
+  }) => {
+    await goToPage(page, pageURLCellGuide);
+
+    // Locate and expand the 'Marker Gene Sets' header if not already expanded
+    let markerGeneSetsHeader = await page.locator(
+      "h5:has-text('Marker Gene Sets')"
+    );
+    let chevronDownIcon = markerGeneSetsHeader.locator(
+      "svg[data-icon='chevron-down']"
+    );
+    if ((await chevronDownIcon.count()) === 0) {
+      await markerGeneSetsHeader.click();
+    }
+
+    await deleteGeneset("enteric smooth muscle cell - marker genes", page);
+
+    // Refresh the page
+    await page.reload();
+
+    // Locate and expand the 'Marker Gene Sets' header if not already expanded
+    markerGeneSetsHeader = await page.locator(
+      "h5:has-text('Marker Gene Sets')"
+    );
+    chevronDownIcon = markerGeneSetsHeader.locator(
+      "svg[data-icon='chevron-down']"
+    );
+    if ((await chevronDownIcon.count()) === 0) {
+      await markerGeneSetsHeader.click();
+    }
```

Comment:
same code as above, let's extract this as a function to keep the code DRY 🙏 

Also might be worth using `tryUntil()` to check that after `await markerGeneSetsHeader.click();` the app is in the expected state (i.e., `(await chevronDownIcon.count()) === 1` ), in case that the click is not registered

Reply:
Wrapping it in a tryUntil caused the test to bug out, not sure why, couldn't figure it out.